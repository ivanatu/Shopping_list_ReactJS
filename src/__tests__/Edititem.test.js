import React from 'react';
import { shallow, render, mount } from 'enzyme';
import ReactDOM from 'react-dom' 
import renderer from 'react-test-renderer'
import Edititem from '../components/edititem'
import toJson from 'enzyme-to-json';
import '../setupTests';



describe('<Edititem/>', () => {
    const wrapper = shallow( <Edititem/> );
    
    it('has a valid snapshot', () => {
        const component = renderer.create(
        <Edititem />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders without crashing', () => {
        render(<Edititem/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="modal"/>))
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="modal-dialog"/>))
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="modal-content"/>))
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="modal-header"/>))
    });
    it('should render a button', () =>{
        expect(wrapper.find('button').length).toEqual(2)
    })
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="modal-body"/>))
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="form-group"/>))
    });
    it('render the inputs', () =>{
        expect(wrapper.find('form').length).toEqual(1);
        expect(wrapper.find('input').length).toEqual(2)
    });
    it('input should respond to change event and change the state', () => {
        wrapper.find('#name').simulate('change', { target: { name: 'name', value: 'apples' } });
        expect(wrapper.state().name).toEqual('apples')
    });
    it('input should respond to change event and change the state', () => {
        wrapper.find('#price').simulate('change', { target: { name: 'price', value: '20' } });
        expect(wrapper.state().price).toEqual('20')
    });
    it('renders the edit item form and submits data', () =>{
        wrapper.setState({name:'apples', price:'20'});
        wrapper.find("#edititem").simulate('submit', {preventDefault(){}});
        expect(toJson(wrapper)).toMatchSnapshot();
        // expect(handleSubmit.calledOnce).toEqual(false)
    });
});