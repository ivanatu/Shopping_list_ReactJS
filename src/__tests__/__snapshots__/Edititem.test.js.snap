// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Edititem/> has a valid snapshot 1`] = `
<div>
  <div
    className="modal"
    id="myModalsundefined"
    role="dialog"
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h4>
            Edit Item
          </h4>
          <button
            className="close"
            data-dismiss="modal"
            id="closeundefined"
            type="button"
          >
            Ã—
          </button>
        </div>
        <div
          className="modal-body"
        >
          <form
            className="register-form"
            method="POST"
            name="login_form"
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              NEW ITEM NAME:
              <input
                className="form-control"
                name="name"
                onChange={[Function]}
                required={true}
                type="text"
                value={undefined}
              />
            </div>
            <div
              className="form-group"
            >
              NEW ITEM PRICE:
              <input
                className="form-control"
                min="0"
                name="price"
                onChange={[Function]}
                required={true}
                step="1"
                type="text"
                value={undefined}
              />
            </div>
            <button
              className="btn btn-info "
              type="submit"
            >
              Save Item
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
